### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by AVyha**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -2,4 +2,55 @@ class Cargo:
     def __init__(self, weight):
         self.weight = weight
 
-# write your code here
+
+class BaseRobot(Cargo):
```
Comment (wQuelS) [inline app/main.py:6]: Does BaseRobot class really need to inherite from Cargo?

**Review by wQuelS — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (AVyha) [reply app/main.py:6]: i think really need because in readme write something like this
image

Comment (wQuelS) [reply app/main.py:6]: Ok, i guess task is not that clear. Mentor will answer that question anyway:)

**Review by wQuelS — APPROVED** [review]

File: app/main.py (hunk 1 above)
Comment (AVyha) [reply app/main.py:6]: Thanks for the review

Comment (Nattalli) [reply app/main.py:6]: @wQuelS is right. That class should not be a child of `Cargo`. The task is to create 3 classes (it is their general count) and use the inheritance (but it doesn't mean that every class should use the inheritance)

File: app/main.py [hunk 2]
```diff
@@ -1,5 +1,59 @@
 class Cargo:
-    def __init__(self, weight):
+    def __init__(self, weight: int) -> None:
         self.weight = weight
 
-# write your code here
+
+class BaseRobot(Cargo):
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
```
Comment (Nattalli) [inline app/main.py:7]: it's better to use `list[int]` here

File: app/main.py [hunk 3]
```diff
@@ -1,5 +1,59 @@
...
+
+    def get_info(self) -> str:
+        return f'Robot: {self.name}, Weight: {self.weight}'
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(self,
+                 name: str,
+                 weight: int,
+                 coords: list = None) -> None:
```
Comment (Nattalli) [inline app/main.py:32]: it's better to use `list[int]` here

File: app/main.py [hunk 4]
```diff
@@ -1,5 +1,59 @@
...
+        self.coords[2] -= step
+
+
+class DeliveryDrone(FlyingRobot):
+    def __init__(self,
+                 name: str,
+                 weight: int,
+                 max_load_weight: int,
+                 current_load: Cargo = None,
+                 coords: list = None) -> None:
```
Comment (Nattalli) [inline app/main.py:49]: it's better to use `list[int]` here

**Review by Nattalli — CHANGES_REQUESTED** [review]

**Review by Nattalli — APPROVED** [review]
Well done! 👍🏻

**Pull request closed without merging**